### ISSUE TITLE
Loading data while training is very slow

### ISSUE THREAD

**Issue by SmashChen** [autonomousvision/transfuser#154]
Many thanks for your contributions. I want to ask you a question. During training, I found that Dataloader load data very slowly, causing the GPU utilization to remain at 0% for a long time. However, despite this, there is still a lot of CPU idle. I tried mimicking the "Cache" module in your code, but that didn't work. If you know a way to solve this problem and let me know, I will be very grateful.

**Comment by Kait0** [comment]
Loading data from the disk is a bottleneck in this code because every sample needs to open many input and label files.
We use Pytorch data workers (cpu threads) to parallelize dataloading and store our data on fast SSD storage using the cache module.
For our 2080ti machines the code is roughly equally IO and compute bound.
If you have faster GPUs your code will likely be IO bound as you mentioned.

I don't have a simple solution for this.
If you have enough RAM you could try loading and keeping the whole dataset in RAM.
I saw some repos using sophisticated database solutions to speed up data loading but I have no experience with that.

If you find a way to speed up data loading signficantly it would be nice if you post it here.
Would also be interesting for me.
